src/utils/convertGrpcEvent.ts

```diff
@@ -86,6 +86,16 @@ const safeParseJSON = <T = unknown>(
   }
 };
 
+const parseMetadata = (grpcRecord: GRPCRecordedEvent, id: string) => {
+  const metadata = grpcRecord.getCustomMetadata_asU8();
+  if (!metadata.length) return;
+  try {
+    return JSON.parse(Buffer.from(metadata).toString("binary"));
```

Comment:
**QUESTION:** Shouldn't we verify somehow if metadata should be in the JSON format? In case it was malformed Json then it will be treated as binary.

Reply:
As discussed, there currently isn't any way of knowing.